### PR TITLE
Add DocumentHTMLPresenter class

### PR DESCRIPTION
**Pull request by seanh**
This isn't used yet but it will be used by the new groups page code in <https://github.com/hypothesis/h/pull/3409>

### CODE REVIEW

**Comment by seanh** [comment]
The diff on this is pretty misleading and GitHub is showing the two commits in the wrong order. What this PR does:

1. Move `AnnotationHTMLPresenter`'s tests into a new `TestAnnotationHTMLPresenter`, without changing the tests.

2. Add new `DocumentHTMLPresenter` class and move several properties from `AnnotationHTMLPresenter` into it.

3. Also move the relevant tests and fixtures from `TestAnnotationHTMLPresenter` into `TestDocumentHTMLPresenter`

4. Leaves behind methods on `AnnotationHTMLPresenter` in place of the ones that got moved into `DocumentHTMLPresenter`, like this:

   ```python
    @property
    def filename(self):
        """Return the filename of this annotation's document."""
        return self.document.filename
   ```

   (`self.document = DocumentHTMLPresenter(self.annotation.document)`)

   Also moved these all to the bottom of the class.

5. Fixed a couple of failing tests in other modules because `AnnotationHTMLPresenter` now requires the `Annotation` to have a `Document`.

I should try getting rid of the backwards-compat methods in 4...

**Comment by seanh** [comment]
> I should try getting rid of the backwards-compat methods in 4...

Hmm, on the other hand it's actually pretty convenient both for the feeds code and its tests if the `DocumentHTMLPresenter` properties are accessible on the `AnnotationHTMLPresenter` object..

File: tests/h/presenters_test.py [hunk 1]
```diff
@@ -5,573 +5,335 @@
...
-    filename.return_value = ""  # This is not a file:// URI.
+        assert annotation.description == (
+            "&lt;blockquote&gt;selected text&lt;/blockquote&gt;entered text")
 
-    annotation = _annotation(annotation)
-    assert isinstance(annotation.title, unicode)
+    def test_created_day_string_from_annotation(self):
+        annotation = self._annotation(
+            annotation=mock.Mock(
+                created="2015-09-04T17:37:49.517852+00:00"))
```
Comment (chdorner) [inline tests/h/presenters_test.py]: This should be `datetime.datetime.datetime(2015, 9, 4, 17, 37, 49, 517852)` now.

**Comment by chdorner** [comment]
Just one tiny oversight in the tests now that we work with `datetime.datetime` objects instead of ISO8601 strings. Otherwise looks great.

File: tests/h/presenters_test.py (hunk 1 above)
Comment (seanh) [inline tests/h/presenters_test.py]: Done. Thanks @chdorner !

**Comment by chdorner** [comment]
LGTM.

**Pull request merged**